src/neo/IO/Caching/TreeNode.cs

```diff
@@ -14,10 +14,11 @@ namespace Neo.IO.Caching
 {
     public class TreeNode<T>
     {
+        private readonly List<TreeNode<T>> children = new();
+
         public T Item { get; }
         public TreeNode<T> Parent { get; }
-
-        private readonly List<TreeNode<T>> children = new();
+        public IReadOnlyList<TreeNode<T>> Children => children;
```

Comment:
```suggestion
        public IReadOnlyList<TreeNode<T>> Children => children.AsReadOnly();
```

Reply:
`AsReadOnly()` will be called every time. I think it's not necessary.